ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/SettlementAssert.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.cmc.ccd.assertion;
+
+import org.assertj.core.api.AbstractAssert;
+import uk.gov.hmcts.cmc.ccd.domain.offers.CCDPartyStatementArrayElement;
+import uk.gov.hmcts.cmc.ccd.domain.offers.CCDSettlement;
+import uk.gov.hmcts.cmc.domain.models.offers.PartyStatement;
+import uk.gov.hmcts.cmc.domain.models.offers.Settlement;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.cmc.ccd.assertion.Assertions.assertThat;
+
+
+public class SettlementAssert extends AbstractAssert<SettlementAssert, Settlement> {
+
+    public SettlementAssert(Settlement actual) {
+        super(actual, SettlementAssert.class);
+    }
+
+    public SettlementAssert isEqualTo(CCDSettlement ccdSettlement) {
+        isNotNull();
+
+        assertThat(actual.getPartyStatements().size()).isEqualTo(ccdSettlement.getPartyStatements().size());
+
+        actual.getPartyStatements()
+            .forEach(partyStatement -> assertPartyStatement(partyStatement, ccdSettlement.getPartyStatements()));
```

Comment:
I think this does not check for the order of party statements. Both lists should have the same party statements in the same order.

Reply:
added a test

Reply:
This will not check if party statements correspond to each other on the lists -- it will look if the CCD list contains statements from the actual list, but the order is not verified. The code should be something like (this is after asserting that lengths match):
```java
for (int i = 0; i < actual.getPartyStatements().getSize(); i++) {
  assertThat(actual.getPartyStatements().get(i)).isEqualTo(ccdSettlement.getPartyStatements().get(i));
}
```